addons/web/static/src/core/domain.js

```diff
@@ -139,15 +139,15 @@ function matchDomain(record, domain) {
   if (domain.length === 0) {
     return true;
   }
-  switch (domain[0]) {
+  const condition = domain.shift();
```

Comment:
why not
```js
const condition = domain[0];
domain = domain.slice(1);
```?
not doing this means that you modify domain in place (after inspection, it looks ok, but it seems dangerous)